src/main/kotlin/no/nav/familie/ef/iverksett/iverksettingstatus/tilstand/TilstandDbUtil.kt

```diff
@@ -0,0 +1,49 @@
+package no.nav.familie.ef.iverksett.iverksettingstatus.status.tilstand
+
+import no.nav.familie.ef.iverksett.iverksetting.domene.DistribuerVedtaksbrevResultat
+import no.nav.familie.ef.iverksett.iverksetting.domene.IverksettResultat
+import no.nav.familie.ef.iverksett.iverksetting.domene.JournalpostResultat
+import no.nav.familie.ef.iverksett.iverksetting.domene.OppdragResultat
+import no.nav.familie.ef.iverksett.iverksetting.domene.TilkjentYtelse
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+@Component
+class TilstandDbUtil(val tilstandJdbc: TilstandJdbc) {
```

Comment:
Litt uklart om denne klassen tilhører domenelaget eller ikke. I så fall, hvis den gjør det, så burde man kanskje bruke et interface pga "separation of concerns" (for å tydeliggjøre eierskap osv, og for å separere infrastruktur fra domenelaget) ? Hvis ikke, så kunne man slettet denne og flyttet transactional til jdbc implementasjonen ? Det føles kanskje ikke ut som en utilities klasse heller ?

Reply:
Er ikke så lett å kategorisere de klassene, fordi det gjøres jo veldig lite akkurat nå. Kanskje litt lettere å ta stilling til det i de tilfellene man ser det trengs et overbygg til repo/jdbc-klassene.
Heller litt mot å slette det mellomlaget, siden de metodene som finnes ikke inneholder noe ekstra logikk. Eneste var vel at vi ønsket å holde transactional-annotasjonen unna jdbc-klassene? Kanskje det heller kunne vært en slags wrapper klasse for de metodene som skal annoteres med transactional, og at klassen heter noe med transactional?

Reply:
Ja, for hvis vi setter transactional på den jdbc metoden, og kaller denne fra den task-klassen, så blir det noe spring-proxy problem ? Eller blir denne transactional hvis vi kaller direkte på den fra den task klassen ?

Reply:
Den blir transactional.

Reply:
Hvorfor må vi ha "TilstandDbUtil" da ? Kan vi ikke interface til jdbc implementasjonen direkte ?

Reply:
Fjernet dbUtil-klassene